enhancements/cluster-logging/azure_workload_identity_auth.md

```diff
@@ -0,0 +1,264 @@
+  //
+  // +kubebuilder:validation:Optional
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="DCR Immutable ID",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  DcrImmutableId string `json:"dcrImmutableId,omitempty"`
+
+  // The stream in the DCR that should handle the custom data
+  // 
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#input-streams
+  //
+  // +kubebuilder:validation:Optional
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Stream Name",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  StreamName string `json:"streamName,omitempty"`
+}
+```
+
+Authentication
+
+```Go
+type AzureMonitorAuthentication struct {
+  // Type is the type of Azure authentication to configure.
+  //
+  // Valid types are:
+  //   1. sharedKey
+  //   2. workloadIdentity
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authentication Type"
+  Type AzureAuthType `json:"type"`
+
+  // Token specifies a bearer token to be used for authenticating requests.
+  //
+  // +nullable
+  // +kubebuilder:validation:Optional
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Bearer Token"
+  Token *BearerToken `json:"token,omitempty"`
+
+  // ClientId points to the secret containing the client ID used for authentication.
+  // 
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#input-streams
+  //
+  // +kubebuilder:validation:Optional
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Client ID",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  ClientId *SecretReference `json:"clientId,omitempty"`
+
+  // TenantId points to the secret containing the tenant ID used for authentication.
+  // 
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#input-streams
+  //
+  // +kubebuilder:validation:Optional
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Stream Name",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  TenantId *SecretReference `json:"tenantId,omitempty"`
+
+  // SharedKey points to the secret containing the shared key used for authenticating requests.
+  //
+  // +nullable
+  // +kubebuilder:validation:Optional
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Shared Key"
+  SharedKey *SecretReference `json:"sharedKey,omitempty"`
+}
+```
+
+- `TenantId` and `ClientId` can be found from the generated [CCO utility secret](#cco-utility-secret) when Openshift is set up for [workload identity for Azure](https://github.com/openshift/cloud-credential-operator/blob/9c3346aea5a7f9a38713c09d11605b8ee825446c/docs/azure_workload_identity.md).
+
+#### Additional configuration fields for the Azure Logs Ingestion sink to `Vector` API
```

Comment:
Are these in addition to those for the upstream PR?

Reply:
Yes. But we might not need `credential_kind` if updating to the latest `azure_identity` sdk, `v0.23.0`